sinks/sinks.go

```diff
@@ -18,6 +18,12 @@ const MetricKeyMetricFlushDuration = "sink.metric_flush_total_duration_ns"
 // place to do this.
 const MetricKeyTotalMetricsFlushed = "sink.metrics_flushed_total"
 
+// DerivedMetricsProcessor processes any metric created from events or service checks into
+// the worker channels for flushing
+type DerivedMetricsProcessor interface {
+	SendMetrics(metrics []samplers.UDPMetric)
+}
+
```

Comment:
One other way to write this, I think, would be to pull the functionality for injecting samples into the metrics workers into its own module (possibly in the `samplers` package), and have both the `ssfmetrics` sink and the `signalfx` sink use that module. That way, the dependency graph looks more like a DAG, and I think we could not have to set up this interface here.

Reply:
Yes, a separate module makes sense. Is that what you want me to change? This kind of obviates some of your other comments as the interface will change when I'm not trying to map functionality on top of an existing module and leverage code reuse.